js/main.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+{
+  const allDiv = document.getElementById('allDiv');
+  
+  // 開始ボタンクリック時の処理
+  document.getElementById('startButton').addEventListener('click', e => {
+    // 取得中の画面表示
+    document.getElementById('title').textContent = '取得中';
+    document.getElementById('comment').textContent = '少々お待ちください';
+    document.getElementById('startButton').remove();
+    // APIからJSONデータを取得
+    fetch("https://opentdb.com/api.php?amount=10")
+    .then(response => {
+      return response.json();
+    })
+    .then(jsonObj => {  
+      const currentCorrectCount = 0; // 正解数カウント
+      let currentQuizNum = 1; // 初期値
+      quiz(jsonObj, currentQuizNum, currentCorrectCount);
+    });
+  });
+
+  // 解答ボタン生成の関数を定義
+  function createQuizButton(textAnswer, correctFlag, quizNum, correctCount, obj) {
+    const button = document.createElement('button'); // ボタンの要素ノードを作成
+    button.textContent = textAnswer.replace(/&#039;/g, '\'').replace(/&quot;/g, '"'); // 特定文字をおきかえる（gは全ての文字列を検索）
+    button.setAttribute('id', correctFlag); // 正解or不正解のidを付与
+    // click時の正解カウントUP処理と次の問題へ移る処理を付与
+    button.addEventListener('click', e => {
+      console.log(e.target.id); // for debug
+      if (e.target.id === "correct") {
+        correctCount ++ ; // 正解数をカウントUP
+      }
+      if (quizNum <= 10) {
+        quiz(obj, quizNum, correctCount); // 次の問題を表示
+      } else {
+        quizResult(correctCount); // クイズの結果を表示
+      };
+    });
+    const pButton = document.createElement('p'); // 解答ボタンを縦に並べるためp要素ノードを作成
+    pButton.appendChild(button); // pにボタンを追加
+    return pButton;
+  };
+
+  // 問題の表示
+  function quiz(obj, quizNum, correctCount) {
```

Comment:
コメントで補わないとわからないメソッド名 => そもそもメソッド名を見直した方がいいことが多いです。
メソッド名は「名詞」ではなく「自動詞」「他動詞 + 目的語」で命名してあげるわかりやすく名前にしやすいです。

```suggestion
  function showQuiz(obj, quizNum, correctCount) {
```

Reply:
修正しました．